graphics/src/AssimpLoader.cc

```diff
@@ -177,16 +182,50 @@ void AssimpLoader::Implementation::RecursiveCreate(const aiScene* _scene, const
   }
 }
 
+void AssimpLoader::Implementation::RecursiveStoreBoneNames(const aiScene *_scene, const aiNode *_node)
+{
+  if (!_node)
+    return;
+
+  for (unsigned meshIdx = 0; meshIdx < _node->mNumMeshes; ++meshIdx)
+  {
+    auto assimpMeshIdx = _node->mMeshes[meshIdx];
+    auto assimpMesh = _scene->mMeshes[assimpMeshIdx];
+    for (unsigned boneIdx = 0; boneIdx < assimpMesh->mNumBones; ++boneIdx)
+    {
+      auto bone = assimpMesh->mBones[boneIdx];
+      auto boneName = std::string(bone->mName.C_Str());
+      gzdbg << "\t" << "RecursiveStoreBoneNames: Bone name: " << boneName << std::endl;
+      BoneNames.push_back(boneName);
+    }
+  }
+
+  // Iterate over children
+  for (unsigned childIdx = 0; childIdx < _node->mNumChildren; ++childIdx)
+  {
+    auto child_node = _node->mChildren[childIdx];
+    // Finally recursive call to explore subnode
+    this->RecursiveStoreBoneNames(_scene, child_node);
+  }
+}
+
 void AssimpLoader::Implementation::RecursiveSkeletonCreate(const aiNode* _node, SkeletonNode* _parent, const math::Matrix4d& _transform)
 {
   // First explore this node
   auto nodeName = ToString(_node->mName);
   // TODO check if node or joint?
-  auto skelNode = new SkeletonNode(_parent, nodeName, nodeName, SkeletonNode::JOINT);
-  // Calculate transform
+  auto boneExist = std::find(BoneNames.begin(), BoneNames.end(), nodeName) != BoneNames.end();
   auto nodeTrans = this->ConvertTransform(_node->mTransformation);
-  skelNode->SetTransform(nodeTrans);
+  auto skelNode = _parent;
```

Comment:
Just thought I'd check whether the logic (and my understanding) are correct:

* If the bone exists, same workflow as before, create a node then recurse through its children setting the current node as parent.
* If the bone does not exist, don't create the node and recurse through the children using the current node and its transformation as a parent

So I guess we will just "skip" ahead while keeping the transforms? Again this sounds reasonable to me but just thought I'd check if my understanding is correct

Reply:
Yes exactly.